pkg/operator/resource/resourceapply/generic.go

```diff
@@ -195,6 +197,18 @@ func ApplyDirectly(ctx context.Context, clients *ClientHolder, recorder events.R
 			} else {
 				result.Result, result.Changed, result.Error = ApplyStorageClass(ctx, clients.kubeClient.StorageV1(), recorder, t)
 			}
+		case *admissionregistrationv1.ValidatingWebhookConfiguration:
+			if clients.kubeClient == nil {
+				result.Error = fmt.Errorf("missing kubeClient")
+			} else {
+				result.Result, result.Changed, result.Error = ApplyValidatingWebhookConfiguration(ctx, clients.kubeClient.AdmissionregistrationV1(), recorder, t, -1)
```

Comment:
@slaskawi this `-1` for `expectedGeneration` means that every time this function is called (i.e., during the static controller sync), the resource will be updated in the apiserver. Is that what we want?

Coincidentally I was working on this and I was thinking about this this to solve this issue:

https://github.com/openshift/library-go/compare/master...bertinatto:webhook-config-static?expand=1#

Reply:
Also, I believe this won't work without registering the scheme in the `init()` function above.

CC @stlaz

Reply:
> Also, I believe this won't work without registering the scheme in the init() function above.

@bertinatto I believe you're right. Let me send an update in a few minutes.

>  this -1 for expectedGeneration means that every time this function is called (i.e., during the static controller sync), the resource will be updated in the apiserver. Is that what we want?
> 
> Coincidentally I was working on this and I was thinking about this this to solve this issue:
>
> https://github.com/openshift/library-go/compare/master...bertinatto:webhook-config-static?expand=1#

Yes, I was aware of that. Unfortunately I couldn't come up with anything better. 

So I decided to do a tradeoff here and always update the resource. Otherwise, a user could modify it and we would never revert his changes.

Perhaps @stlaz has a better idea how to solve this?

Reply:
PR added: https://github.com/openshift/library-go/pull/1135

Reply:
PR#1135 does not fixes the issue @bertinatto pointed out though right?

Reply:
No. The code as it is now always updates performs an update.